sdk/storage/examples/table_client_00.rs

```diff
@@ -0,0 +1,178 @@
+use azure_core::{Context, Error};
+use azure_storage::table::prelude::*;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct UserEntity {
+    #[serde(rename = "PartitionKey")]
+    pub city: Option<String>,
+    #[serde(rename = "RowKey")]
+    pub surname: Option<String>,
+    pub name: Option<String>,
+}
+
+impl UserEntity {
+    pub fn new(city: &str, surname: &str, name: &str) -> Self {
+        UserEntity {
+            name: Some(name.into()),
+            city: Some(city.into()),
+            surname: Some(surname.into()),
+        }
+    }
+}
+
+impl<'a> TableEntity<'a> for UserEntity {
+    type Entity = Self;
+
+    fn partition_key(&self) -> &str {
+        self.city.as_ref().unwrap().as_str()
+    }
+
+    fn row_key(&self) -> &str {
+        self.surname.as_ref().unwrap().as_str()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct UserEntityExtended<'a> {
+    #[serde(rename = "PartitionKey")]
+    pub city: &'a str,
+    #[serde(rename = "RowKey")]
+    pub surname: &'a str,
+    pub age: u8,
+}
+
+impl<'a> TableEntity<'a> for UserEntityExtended<'a> {
+    type Entity = Self;
+
+    fn partition_key(&self) -> &str {
+        self.city
+    }
+
+    fn row_key(&self) -> &str {
+        self.surname
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    // in the following example we will interact with the user table. first let's create the table if it doesn't exists;
+    let users = vec![
+        UserEntity::new("beit dagan", "shem tov", "or"),
+        UserEntity::new("rishon lezion", "gerbil", "yaron"),
+        UserEntity::new("poria neve oved", "sachanov", "shay"),
+    ];
+
+    let table_name = "users";
+    let table_client = TableClient::emulator(TableOptions::default());
```

Comment:
I'd rather give the ability to test against Azure as well. In the SDK we have the unwritten convention to pass storage account and master key with env variables.

Something like:

```rust
    let account = std::env::vars()
        .find(|i| i.0 == "STORAGE_ACCOUNT")
        .map(|i| i.1);
    let key = std::env::vars()
        .find(|i| i.0 == "STORAGE_MASTER_KEY")
        .map(|i| i.1);

    let auth_token = match (account.as_ref(), key) {
        (Some(account), Some(key)) => Some(AuthorizationToken::SharedKeyToken {
            account: account.to_owned(),
            key,
        }),
        _ => None,
    };

    let table_name = "users";
    let table_client = auth_token
        .as_ref()
        .map(|auth_token| {
            TableClient::new(
                account.clone().unwrap(),
                auth_token.clone(),
                TableOptions::default(),
            )
        })
        .unwrap_or_else(|| TableClient::emulator(TableOptions::default()));
```